indico/modules/events/registration/client/js/form/FormErrorList.jsx

```diff
@@ -0,0 +1,86 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React from 'react';
+import {useFormState} from 'react-final-form';
+import {useSelector} from 'react-redux';
+
+import {Translate} from 'indico/react/i18n';
+
+import {getFieldLabelLookup} from './selectors';
+import './FormErrorList.module.scss';
+
+export default function FormErrorList() {
+  const fieldLabelLookup = useSelector(getFieldLabelLookup);
+  const {errors} = useFormState({errors: true});
+
+  const fieldsWithErrors = [];
+  for (const fieldName in errors) {
+    const field = fieldLabelLookup.get(fieldName);
+
+    // Check for misbehaving fields. This is for developers.
+    if (!field) {
+      console.warn(
+        `FormErrorList: Field ${fieldName} is not registered with the final form and will not be shown in the error list`
+      );
+      continue;
+    }
+
+    fieldsWithErrors.push({...field, message: errors[fieldName]});
+  }
+
+  function focusField(evt) {
+    evt.preventDefault();
+    const id = evt.target.href.split('#')[1];
+    const element = document.getElementById(id);
+
+    if (!element) {
+      // This is a programmer error. All fields *must* have an id.
+      console.warn(
+        `FormError: #${id} does not match a form field. Please make sure all form fields have an appropriate id.`
```

Comment:
There are some more fields that do not have an ID. I recommend creating a registration form that has one field of each type to test this.

For certain field types it's not really clear how beneficial the ID would be though, for example radio buttons - but the ID could only target its container since targeting any specific radio button would immediately select it. Might still be useful for keyboard navigation though?

Some field types where this happens:

- Yes/No
- Accommodation
- Single choice (w/ radio buttons)

I have not tested all the other fields though.

Reply:
In this case, the ID is basically a crude approximation. Ideally I'd love to have been using the built-in constraint validation which would obviate the need for this mechanism to begin with, and it supports assigning anchor elements to errors, so that the browser knows where to jump. Alas. 🥲

Anyway, I'll check out what different fields do and come up with a better design.

Reply:
I think I've addressed this and most other issues. Now, I still can't test the behavior of the new date picker and how that integrates into this error message system, but I can address that in either PR so your call.